visualization/dashboard/src/MLIDashboard/Controls/CohortEditor/CohortEditor.tsx

```diff
@@ -372,14 +417,59 @@ export class CohortEditor extends React.PureComponent<ICohortEditorProps, ICohor
     }
 
     private setFilterLabel(filter: IFilter): React.ReactNode {
-        //TODO: change the function unwrap neatly 
+        const selectedFilter = this.props.jointDataset.metaDict[filter.column];
         let label = "";
-        const space = " ";
-        label = this.props.jointDataset.metaDict[filter.column].abbridgedLabel + space
-        label += filter.method + space
-        label += filter.arg
+        label = selectedFilter.abbridgedLabel
+
+        const filterMethod = this.getFilterMethodLabel(filter.method)
+            label += filterMethod
+        if (selectedFilter.isCategorical) {
+            let selectedValues = [];
+            let filterArgs = filter.arg as number[];
+            filterArgs.forEach((element) => {
+                selectedValues.push(selectedFilter.sortedCategoricalValues[element])
+            }
+            );
+            label += selectedValues;
+        }
+        else {
+            label += filter.arg
+        }
+
+        if (filter.method == FilterMethods.inTheRangeOf) {
+            label += localization.FilterOperations.inTheRangeOfClose
+        }
+
+        return (<TooltipHost
+        overflowMode={TooltipOverflowMode.Self}
+        hostClassName={styles.filterLabel}
+        content={label}
+        onTooltipToggle={() => false}
+        styles={tooltip}
+      >
+        {label}
+      </TooltipHost>);
 
-        return (<Text variant={"small"} className={styles.filterLabel}>{label}</Text>);
+    }
+
+    private getFilterMethodLabel(filterMethod: FilterMethods): string {
+        let label = "";
```

Comment:
An alternate approach could be to use a dictionary, where the key is a filterMethod, and the value is the localized string. This type can be defined in Typescript as const dictOfLabels: { [key: string]: string} = {FilterMethod.foo: localize.foo, ...}; This could be defined as a private static property on this class, or on the FilterMethods class since either could make a claim to owning that dictionary.

Reply:
the string returned could be a template (in general) so that inTheRange could be "in the range of {0} and {1}" or "[{0}, {1}]" however it was defined. The argument for the numeric operations may need to be formatted (to prevent numbers with lots of digits).

Reply:
added a method roundDecimalValue which will handle decimal numbers.